task.ps1

```diff
@@ -42,5 +42,18 @@ New-AzVm `
 -SubnetName $subnetName `
 -VirtualNetworkName $virtualNetworkName `
 -SecurityGroupName $networkSecurityGroupName `
--SshKeyName $sshKeyName 
-# -PublicIpAddressName $publicIpAddressName
+-SshKeyName $sshKeyName `
+-Zone 1
```

Comment:
When specifying the zone for the VM, ensure that the chosen zone is available in the specified location (`$location`). This is important to avoid deployment errors.